log/term/colorlogger.go

```diff
@@ -0,0 +1,149 @@
+package term
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+)
+
+// Color is the abstract color, the zero value is the Default.
+type Color uint8
+
+const (
+	NoColor = Color(iota)
+	Red
+	Green
+	Yellow
+	Blue
+	Magenta
+	Cyan
+	White
+	Default
+
+	maxColor
+)
+
+var resetColorBytes = []byte("\x1b[0m")
+var fgColorBytes [][]byte
+var bgColorBytes [][]byte
+
+func init() {
+	for color := NoColor; color < maxColor; color++ {
+		fgColorBytes = append(fgColorBytes, []byte(fmt.Sprintf("\x1b[%dm", 30+color)))
+		bgColorBytes = append(bgColorBytes, []byte(fmt.Sprintf("\x1b[%dm", 40+color)))
+	}
+}
+
+type FgBgColor struct {
+	Fg, Bg Color
+}
+
+func (c FgBgColor) IsZero() bool {
+	return c.Fg == NoColor && c.Bg == NoColor
+}
+
+// NewColorLogger returns a log.Logger which writes colored logs to w. It
+// colors whole records based on the FgBgColor returned by the color function.
+// Log events are formatted by the Logger returned by newLogger.
+func NewColorLogger(w io.Writer, newLogger func(io.Writer) log.Logger, color func(keyvals ...interface{}) FgBgColor) log.Logger {
+	if color == nil {
+		panic("color func nil")
+	}
```

Comment:
I'm not sure about that. By constructing a ColorLogger, the user is implicitly declaring that they intend to read its output on a TTY. I don't think a further if-guard on that intent is appropriate; it makes sense to me to bake that assumption into the constructor. And, taking it a step further, if that assumption is invalidated at runtime, it would be incidental, and as a user I'd expect the logger to do the right thing.

What do you think?